### PR TITLE
[Docker] Pin fastapi in a separate requirements file

### PR DESCRIPTION
**Pull request by amogkam**
Signed-off-by: Amog Kamsetty [amogkamsetty@yahoo.com](mailto:amogkamsetty@yahoo.com)

The ML Docker image does not pick up changes in requirements.txt due to a bug. This is being fixed here https://github.com/ray-project/ray/pull/29375.

We pin fastapi in a separate file as a stopgap solution to immediately resolve https://github.com/ray-project/ray/issues/29396

Closes https://github.com/ray-project/ray/issues/29396

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by simon-mo — APPROVED** [review]

File: python/requirements/requirements_extra.txt
Comment (rickyyx) [inline python/requirements/requirements_extra.txt:1]: Since https://github.com/ray-project/ray/pull/29375 is also a release blocker - should we wait until that is merged and resolved this? 
cc @krfricke

**Pull request merged**
